nion/eels_analysis/BackgroundModel.py

```diff
@@ -119,8 +120,41 @@ def __unused_perform_fit(self, xs: numpy.ndarray, ys: numpy.ndarray, fs: numpy.n
         series = typing.cast(typing.Any, numpy.polynomial.polynomial.Polynomial.fit(xs, transform_data(ys), self.deg))
         return untransform_data(series(fs))
 
+class TwoAreaBackgroundModel(AbstractBackgroundModel):
+
+    def __init__(self, background_model_id: str, params_func = None, model_func = None, title: str = None):
+        super().__init__(background_model_id, title)
+        self.model_func = model_func
+        self.params_func = params_func
+
+
+    def _perform_fits(self, xs: numpy.ndarray, yss: numpy.ndarray, fs: numpy.ndarray) -> numpy.ndarray:
+        if not len(xs) % 2 == 0:
+            yss = yss[...,:-1]
+            xs = xs[:-1]
+        half_interval = int(len(xs)/2)
+        areas_1 = numpy.trapz(yss[...,:half_interval], xs[:half_interval], axis=1)
+        areas_2 = numpy.trapz(yss[...,half_interval:], xs[half_interval:], axis=1)
+        x_start = xs[0]
+        x_center = xs[half_interval]
+        x_end = xs[-1]
+        p1s, p2s = self.params_func(areas_1,areas_2, x_start, x_center, x_end)
+        series = numpy.empty((len(yss),len(fs)))
+        for idx, (p1, p2) in enumerate(zip(p1s,p2s)):
+            series[idx] = self.model_func(fs,p1,p2)
+        return series
+
+def power_law_params(area_1,area_2, x_start, x_center, x_end):
+    r = 2 * (numpy.log(area_1) - numpy.log(area_2)) / (numpy.log(x_end) - numpy.log(x_start))
+    k = 1 - r
+    A = k * area_2 / (x_end ** k - x_center ** k)
+    return A, r
+
+def power_law_func(x,A,r):
```

Comment:
Same: typing.

def power_law_func(x: numpy.ndarray, A: float, r: float) -> float:

(actually not sure if 'x' is a float or an ndarray...? that's why I ask for typing)